crates/router/src/core/payments/helpers.rs

```diff
@@ -1949,7 +2010,8 @@ pub(crate) fn validate_payment_method_fields_present(
     utils::when(
         req.payment_method.is_some()
             && req.payment_method_data.is_none()
-            && req.payment_token.is_none(),
+            && req.payment_token.is_none()
+            && req.recurring_details.is_none(),
```

Comment:
Why is this required?

Reply:
Now that i'm getting the `payment_method` from `payment_method_info` itself as suggested by you, we can remove this